improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -510,6 +541,35 @@ def __init__(
         self.mask_percentiles = mask_percentiles
         self.skip_ecc_bounds = skip_ecc_bounds
 
+    @staticmethod
+    def _assess_if_ecc_bounds_needed(
+        forecast_probabilities: Cube, threshold_points: ndarray, threshold_name: str
+    ) -> bool:
+        """
+        Function to check whether ECC bounds are required for the percentile generation
+        process. ECC bounds are only necessary if the largest threshold defined on a
+        cube has non-zero probability of being exceeded.
+
+        Args:
+            forecast_probabilities:
+                Cube with a threshold coordinate.
+            threshold_points:
+                Array of threshold values used to calculate the probabilities.
+            threshold_name:
+                Name of the threshold coordinate.
+
+        Returns:
+            Boolean indicating whether ECC bounds are needed (True if needed, False if
+            not).
+        """
+        largest_threshold_slice = forecast_probabilities.extract(
+            iris.Constraint(**{threshold_name: threshold_points[-1]})
+        )
+        if np.any(largest_threshold_slice.data != 0):
+            return True
+        else:
+            return False
```

Comment:
Can also be a one-liner
```suggestion
        return np.any(largest_threshold_slice.data != 0)
```